.github/workflows/tests.yml

```diff
@@ -320,7 +320,11 @@ jobs:
         echo "$WHEEL_FPATH=WHEEL_FPATH"
         echo "$INSTALL_EXTRAS=INSTALL_EXTRAS"
         echo "$MOD_VERSION=MOD_VERSION"
-        pip install --prefer-binary "xdoctest[$INSTALL_EXTRAS]==$MOD_VERSION" -f wheelhouse
+        echo "Downloading from pypi"
+        pip download "xdoctest[$INSTALL_EXTRAS]==$MOD_VERSION" --dest wheeldownload
+        echo "Overwriting pypi xdoctest wheel"
+        cp wheelhouse/* wheeldownload/
+        pip install --prefer-binary "xdoctest[$INSTALL_EXTRAS]==$MOD_VERSION" -f wheeldownload --no-index
```

Comment:
Ah, that's a neat trick to work around the issue. I will codify that in xcookie.

Reply:
A question I have with this is does `pip download` use the pip cache if it is available?

I have CI jobs that use these same templates that have much heavier dependencies and having some of them in a cache helps a lot. I'm wondering if this utilizes it or does it ignore it and download every time?

Reply:
Hmm, also thinking about this if there is a version conflict between the CI and pypi, what happens if the CI is testing a new set of dependencies. I would think `pip download "xdoctest[$INSTALL_EXTRAS]==$MOD_VERSION" --dest wheeldownload` will still incorrectly pull the xdoctest wheel from pypi and use whatever dependencies are specified there instead of the ones specified in the wheel built by the CI. So maybe this just pushes the can down the road?

Reply:
pip download will use the cache afaiaa because it has a `--no-cache-dir` option.

You're right about the CI if there are new dependencies.  Is there a reason why the CI doesn't just install the wheel directly without using `--find-links` like this: `pip install "$WHEEL_FPATH[$INSTALL_EXTRAS]"`